Source/Core/OrientedBoundingBox.js

```diff
@@ -0,0 +1,332 @@
+            !defined(maximumZ)) {
+            throw new DeveloperError('all extents (minimum/maximum X/Y/Z) are required.');
+        }
+        //>>includeEnd('debug');
+
+        if (!defined(result)) {
+            result = new OrientedBoundingBox();
+        }
+
+        var halfAxes = result.halfAxes;
+        Matrix3.setColumn(halfAxes, 0, tangentPlane.xAxis, halfAxes);
+        Matrix3.setColumn(halfAxes, 1, tangentPlane.yAxis, halfAxes);
+        Matrix3.setColumn(halfAxes, 2, tangentPlane.zAxis, halfAxes);
+
+        var centerOffset = scratchOffset;
+        centerOffset.x = (minimumX + maximumX) / 2.0;
+        centerOffset.y = (minimumY + maximumY) / 2.0;
+        centerOffset.z = (minimumZ + maximumZ) / 2.0;
+
+        var scale = scratchScale;
+        scale.x = (maximumX - minimumX) / 2.0;
+        scale.y = (maximumY - minimumY) / 2.0;
+        scale.z = (maximumZ - minimumZ) / 2.0;
+
+        var center = result.center;
+        centerOffset = Matrix3.multiplyByVector(halfAxes, centerOffset, centerOffset);
+        Cartesian3.add(tangentPlane.origin, centerOffset, center);
+        Matrix3.multiplyByScale(halfAxes, scale, halfAxes);
+
+        return result;
+    };
+
+    var scratchRectangleCenterCartographic = new Cartographic();
+    var scratchRectangleCenter = new Cartesian3();
+    var perimeterCartographicScratch = [new Cartographic(), new Cartographic(), new Cartographic(), new Cartographic(), new Cartographic(), new Cartographic(), new Cartographic(), new Cartographic()];
+    var perimeterCartesianScratch = [new Cartesian3(), new Cartesian3(), new Cartesian3(), new Cartesian3(), new Cartesian3(), new Cartesian3(), new Cartesian3(), new Cartesian3()];
+    var perimeterProjectedScratch = [new Cartesian2(), new Cartesian2(), new Cartesian2(), new Cartesian2(), new Cartesian2(), new Cartesian2(), new Cartesian2(), new Cartesian2()];
+    /**
+     * Computes an OrientedBoundingBox that bounds a {@link Rectangle} on the surface of an {@link Ellipsoid}.
+     * There are no guarantees about the orientation of the bounding box.
+     *
+     * @param {Ellipsoid} ellipsoid The ellipsoid on which the rectangle is defined.
+     * @param {Rectangle} rectangle The cartographic rectangle on the surface of the ellipsoid.
+     * @param {Number} [minimumHeight=0.0] The minimum height (elevation) within the tile.
+     * @param {Number} [maximumHeight=0.0] The maximum height (elevation) within the tile.
+     * @param {OrientedBoundingBox} [result] The object onto which to store the result.
+     * @returns {OrientedBoundingBox} The modified result parameter or a new OrientedBoundingBox instance if none was provided.
+     *
+     * @exception {DeveloperError} rectangle.width must be between 0 and pi.
+     * @exception {DeveloperError} rectangle.height must be between 0 and pi.
+     */
+    OrientedBoundingBox.fromEllipsoidRectangle = function(ellipsoid, rectangle, minimumHeight, maximumHeight, result) {
```

Comment:
`OrientedBoundingBox.fromRectangle` would be a fine name.

Make the order
```
rectangle, minimumHeight, maximumHeight, ellipsoid, result
```
As I suggested previously, default `ellipsoid` to WGS84 like we do everywhere else.